lib/net/ldap/connection.rb

```diff
@@ -562,12 +562,13 @@ def search(args = nil)
       result_pdu || OpenStruct.new(:status => :failure, :result_code => Net::LDAP::ResultCodeOperationsError, :message => "Invalid search")
     end # instrument
   ensure
+
     # clean up message queue for this search
     messages = message_queue.delete(message_id)
 
     # in the exceptional case some messages were *not* consumed from the queue,
     # instrument the event but do not fail.
-    unless messages.empty?
+    unless messages.nil? or messages.empty?
```

Comment:
This scenario is possible, but it's really hard to test. `message_queue` will always return a hash, but the `delete` can return `nil` if `message_id` isn't in that hash. I tried testing this with stubs, but `queued_read` will loop forever trying to read additional pdf's, so this code is never reached. In real life, the socket would terminate at some point, and this `ensure` block would run, but I don't know how to simulate that in a test.